index.js

```diff
@@ -90,67 +146,59 @@ function triggerInstalls(approvedUpdates) {
   });
 }
 
-function setupWorker() {
-  if (self.loadReason === 'install') {
-    sendToWeb({type: 'addon-self:installed'});
-  } else if (self.loadReason === 'enable') {
-    sendToWeb({type: 'addon-self:enabled'});
-  } else if (self.loadReason === 'upgrade') {
-    sendToWeb({type: 'addon-self:upgraded'});
-  }
-
-  mod.port.on('from-web-to-addon', function(msg) {
-    if (!msg.type) { return; }
-    switch (msg.type) {
-      case 'update-check':
-        sendToWeb({ type: 'addon-updates', detail: state.updates });
-        break;
-      case 'update-approve':
-        state.updates = require('exclude')(state.updates, msg.detail);
-        button.badge = state.updates.length;
-        triggerInstalls(msg.detail);
-        break;
-      case 'loaded':
-        sendToWeb({ type: 'addon-available' });
-        break;
-      default:
-        sendToWeb({ type: 'echo', data: msg });
-        break;
-    }
-  });
-
-  const installListeners = {
-    onInstallEnded: function(install, addon) {
-      sendToWeb({type: 'addon-install:install-ended', detail: formatInstallData(install, addon)});
-    },
-    onInstallFailed: function(install) {
-      sendToWeb({type: 'addon-install:install-failed', detail: formatInstallData(install)});
-    },
-    onInstallStarted: function(install) {
-      sendToWeb({type: 'addon-install:install-started', detail: formatInstallData(install)});
-    },
-    onNewInstall: function(install) {
-      sendToWeb({type: 'addon-install:install-new', detail: formatInstallData(install)});
-    },
-    onInstallCancelled: function(install) {
-      sendToWeb({type: 'addon-install:install-cancelled', detail: formatInstallData(install)});
-    },
-    onDownloadStarted: function(install) {
-      sendToWeb({type: 'addon-install:download-started', detail: formatInstallData(install)});
-    },
-    onDownloadProgress: function(install) {
-      sendToWeb({type: 'addon-install:download-progress', detail: formatInstallData(install)});
-    },
-    onDownloadEnded: function(install) {
-      sendToWeb({type: 'addon-install:download-ended', detail: formatInstallData(install)});
-    },
-    onDownloadCancelled: function(install) {
-      sendToWeb({type: 'addon-install:download-cancelled', detail: formatInstallData(install)});
-    },
-    onDownloadFailed: function(install) {
-      sendToWeb({type: 'addon-install:download-failed', detail: formatInstallData(install)});
-    }
+function formatInstallData(install, addon) {
+  let formatted = {
+    'name': install.name ? install.name : '',
```

Comment:
You may be able to simplify this to just `'name': install.name || '',` (and same w/ `version` below).